test/metabase/db/data_migrations_test.clj

```diff
@@ -331,3 +339,107 @@
           (testing "And it is idempotent"
             (#'migrations/migrate-click-through)
             (is (= expected-settings (get-settings!)))))))))
+
+(def ^:private default-saml-idp-certificate
+  "Public certificate from Auth0, used to validate mock SAML responses from Auth0"
+  "MIIDEzCCAfugAwIBAgIJYpjQiNMYxf1GMA0GCSqGSIb3DQEBCwUAMCcxJTAjBgNV
+  BAMTHHNhbWwtbWV0YWJhc2UtdGVzdC5hdXRoMC5jb20wHhcNMTgwNTI5MjEwMDIz
+  WhcNMzIwMjA1MjEwMDIzWjAnMSUwIwYDVQQDExxzYW1sLW1ldGFiYXNlLXRlc3Qu
+  YXV0aDAuY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzNcrpju4
+  sILZQNe1adwg3beXtAMFGB+Buuc414+FDv2OG7X7b9OSYar/nsYfWwiazZRxEGri
+  agd0Sj5mJ4Qqx+zmB/r4UgX3q/KgocRLlShvvz5gTD99hR7LonDPSWET1E9PD4XE
+  1fRaq+BwftFBl45pKTcCR9QrUAFZJ2R/3g06NPZdhe4bg/lTssY5emCxaZpQEku/
+  v+zzpV2nLF4by0vSj7AHsubrsLgsCfV3JvJyTxCyo1aIOlv4Vrx7h9rOgl9eEmoU
+  5XJAl3D7DuvSTEOy7MyDnKF17m7l5nOPZCVOSzmCWvxSCyysijgsM5DSgAE8DPJy
+  oYezV3gTX2OO2QIDAQABo0IwQDAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBSp
+  B3lvrtbSDuXkB6fhbjeUpFmL2DAOBgNVHQ8BAf8EBAMCAoQwDQYJKoZIhvcNAQEL
+  BQADggEBAAEHGIAhR5GPD2JxgLtpNtZMCYiAM4Gr7hoTQMaKiXgVtdQu4iMFfbpE
+  wIr6UVaDU2HKhvSRFIilOjRGmCGrIzvJgR2l+RL1Z3KrZypI1AXKJT5pF5g5FitB
+  sZq+kiUpdRILl2hICzw9Q1M2Le+JSUcHcbHTVgF24xuzOZonxeE56Oc26Ju4CorL
+  pM3Nb5iYaGOlQ+48/GP82cLxlVyi02va8tp7KP03ePSaZeBEKGpFtBtEN/dC3NKO
+  1mmrT9284H0tvete6KLUH+dsS6bDEYGHZM5KGoSLWRr3qYlCB3AmAw+KvuiuSczL
+  g9oYBkdxlhK9zZvkjCgaLCen+0aY67A=")
+
+(defn- call-with-sso-configured
+  [sso-group-mapping f]
+  (premium-features-test/with-premium-features #{:sso}
+    (mt/with-temporary-setting-values
+      [jwt-enabled                        true
+       jwt-identity-provider-uri          "http://test.idp.metabase.com"
+       jwt-shared-secret                  (crypto-random/hex 32)
+       jwt-group-mappings                 sso-group-mapping
+       saml-enabled                       true
+       saml-identity-provider-uri         "http://test.idp.metabase.com"
+       saml-identity-provider-certificate default-saml-idp-certificate
+       saml-group-mappings                sso-group-mapping]
+      (f))))
+
+(defn- call-with-ldap-configured [ldap-group-mapping f]
+  (mt/with-temporary-setting-values
+    [ldap-enabled          true
+     ldap-host             "http://localhost:8888"
+     ldap-user-base        "dc=metabase,dc=com"
+     ldap-group-mappings   ldap-group-mapping
+     ldap-sync-admin-group false]
+    (f)))
+
+(defmacro ^:private with-ldap-and-sso-configured
```

Comment:
Nice job getting all this stuff working, but especially given my comment above about running the SAML/JWT migration logic even on OSS instance, I think having SSO actually configured in these tests is overkill. 

You should be able to just set the `saml-enabled`/`jwt-enabled` setting to `true` directly in the DB, run the migration and check that stuff was updated correctly, then reset those settings to their prior value. Similar to what `with-temporary-setting-values` does, but I don't think you can use that directly here since those setting definitions live in EE code. (Although I'm not totally sure if that's an issue for using that helper or not)

Reply:
> running the SAML/JWT migration logic even on OSS instance

Actually, it'll not. SSO will only be configured if enterprise is available, check lines 396->399 where it runs the `body` with only `ldap` configured if we run OSS

Is it too much `magic` under the hood though?

Reply:
> Although I'm not totally sure if that's an issue for using that helper or not

It'll throw an exception if we try to access an _enterprise_ in OSS.